plugins/vpp/natplugin/vppcalls/vpp2106/nat_vppcalls.go

```diff
@@ -0,0 +1,782 @@
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vpp2106
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/pkg/errors"
+
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2106/interface_types"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2106/ip_types"
+	vpp_nat_ed "go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2106/nat44_ed"
+	vpp_nat_ei "go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2106/nat44_ei"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2106/nat_types"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/natplugin/vppcalls"
+	nat "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/nat"
+)
+
+// Num protocol representation
+const (
+	ICMP uint8 = 1
+	TCP  uint8 = 6
+	UDP  uint8 = 17
+)
+
+const (
+	// NoInterface is sw-if-idx which means 'no interface'
+	NoInterface = interface_types.InterfaceIndex(^uint32(0))
+	// Maximal length of tag
+	maxTagLen = 64
+)
+
+// holds a list of NAT44 ED flags set
+type nat44EdFlags struct {
+	isTwiceNat     bool
+	isSelfTwiceNat bool
+	isOut2In       bool
+	isAddrOnly     bool
+	isOutside      bool
+	isInside       bool
+	isStatic       bool
+	isExtHostValid bool
+}
+
+// holds a list of NAT44 EI flags set
+type nat44EiFlags struct {
+	eiStaticMappingOnly  bool
+	eiConnectionTracking bool
+	eiOut2InDpo          bool
+	eiAddrOnlyMapping    bool
+	eiIfInside           bool
+	eiIfOutside          bool
+	eiStaticMapping      bool
+}
+
+func (h *NatVppHandler) enableNAT44EdPlugin(opts vppcalls.Nat44InitOpts) error {
+	var flags vpp_nat_ed.Nat44ConfigFlags
+	if opts.ConnectionTracking {
+		flags |= vpp_nat_ed.NAT44_IS_CONNECTION_TRACKING
+	}
+	if opts.StaticMappingOnly {
+		flags |= vpp_nat_ed.NAT44_IS_STATIC_MAPPING_ONLY
+	}
+	if opts.OutToInDPO {
+		flags |= vpp_nat_ed.NAT44_IS_OUT2IN_DPO
+	}
+
+	req := &vpp_nat_ed.Nat44EdPluginEnableDisable{
+		Enable: true,
+		Flags:  flags,
+	}
+	reply := &vpp_nat_ed.Nat44EdPluginEnableDisableReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *NatVppHandler) enableNAT44EiPlugin(opts vppcalls.Nat44InitOpts) error {
+	var flags vpp_nat_ei.Nat44EiConfigFlags
+	if opts.ConnectionTracking {
+		flags |= vpp_nat_ei.NAT44_EI_CONNECTION_TRACKING
+	}
+	if opts.StaticMappingOnly {
+		flags |= vpp_nat_ei.NAT44_EI_STATIC_MAPPING_ONLY
+	}
+	if opts.OutToInDPO {
+		flags |= vpp_nat_ei.NAT44_EI_OUT2IN_DPO
+	}
+
+	req := &vpp_nat_ei.Nat44EiPluginEnableDisable{
+		Enable: true,
+		Flags:  flags,
+	}
+	reply := &vpp_nat_ei.Nat44EiPluginEnableDisableReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Enable NAT44 plugin and apply the given set of options.
+func (h *NatVppHandler) EnableNAT44Plugin(opts vppcalls.Nat44InitOpts) error {
```

Comment:
The comment should have the following format `EnableNAT44Plugin ...`. Also, please mention that this VPP version supports NAT_ED & NAT_EI simultaneously, but vpp-agent does not support it yet. (`// TODO ...`)

Reply:
done